src/ssl.c

```diff
@@ -7899,6 +7922,7 @@ int wolfSSL_DTLS_SetCookieSecret(WOLFSSL* ssl,
             }
 #endif /* NO_HANDSHAKE_DONE_CB */
 
+#ifndef WOLFSSL_WPAS
```

Comment:
The resources are freed because the data is no longer needed. The wpa_supplicant code looks at the values post-handshake, therefore the resources are not freed if building for WPAS.